library/item/item_extension/src/main/java/org/quiltmc/qsl/item/extension/mixin/ProjectileUtilMixin.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.item.extension.mixin;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.ProjectileUtil;
+import net.minecraft.item.Item;
+import net.minecraft.item.Items;
+import net.minecraft.util.Hand;
+import org.quiltmc.qsl.item.extension.api.bow.BowExtensions;
+import org.quiltmc.qsl.item.extension.api.crossbow.CrossbowExtensions;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(ProjectileUtil.class)
+public abstract class ProjectileUtilMixin {
+	private static final Hand[] HANDS = Hand.values(); // Cache the hands to not create the hands array each time the loop is run
```

Comment:
I was worried about the possibility of an additional hand being added and such, but then I realized that it would require enum extension shenanigans

Reply:
now my mod that adds a secret third hand is incompatible with quilt, thanks ennui /j

Reply:
time to make it /j

Reply:
you joke but I've done this before

Reply:
This would be fine since it gets the array after any enum would be added